### PR TITLE
Add Observability guide

### PR DESCRIPTION
**Pull request by antoineco**
Closes triggermesh/docs#253

Live preview available at https://docs-issue-253-antoineco.k.triggermesh.io/guides/observability/

### CODE REVIEW

File: docs/guides/observability-metrics.md [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+software, and for storage in a time-series database. Prometheus itself is a popular choice for this job, as it includes
+both a server for scraping telemetry metrics, and a time-series database optimized for the Prometheus data model. 
+
+![Metrics scraping](../assets/guides/observability/metrics-scrape.excalidraw.png)
+
+Most of the available metrics fall into the categories described previously in this document, whenever applicable based
+on the type of component. Some components may expose additional, application-specific metrics. The data model presented
+in the previous section makes these metrics easily discoverable by consumers.
+
+## Collection and Analysis
```
Comment (jmcx) [inline docs/guides/observability-metrics.md:70]: We could add "with Prometheus and Grafana" to the title to show this section is a practical example?

**Review by jmcx — APPROVED** [review]
Awesome stuff! just one small comment on a header

File: docs/guides/observability-metrics.md (hunk 1 above)
Comment (antoineco) [reply docs/guides/observability-metrics.md]: Addressed in [ca6da7f5..59acc041](https://github.com/triggermesh/docs/compare/ca6da7f5eb00e390c91b11b9256d1493332c074b..59acc04112860984ea0596b416e97db5f3680955)

File: mkdocs.yml [hunk 2]
```diff
@@ -56,6 +56,10 @@ nav:
       - Using an AWS AMI: guides/installation-ami.md
       - Using Helm: guides/installation-helm.md
       - Tanzu Community Edition: guides/installation-tce.md
+    - Observability:
+      - '': guides/observability.md
+      - Structured logging: guides/observability-logging.md
```
Comment (sameersbn) [inline mkdocs.yml:61]: @antoineco no such guide exists in the repo. is it possible you failed to commit it or something?

Comment (antoineco) [reply mkdocs.yml:61]: It was meant as a placeholder because I thought maybe someone would write that one in parallel. We should probably comment this.

**Pull request merged**